### DISCUSSION TITLE
Support for Static Linux SDK Based on Musl Instead of Glibc

### DISCUSSION THREAD

**Discussion by AnneBlair** [krzyzanowskim/CryptoSwift#1057] [Q&A] [answered]
Hello,

I am currently working on a project that requires the use of a static Linux SDK. We are opting to use Musl instead of Glibc due to its excellent support for static linking and its permissive licensing, which simplifies the distribution of statically linked executables.

To ensure compatibility with Musl, I suggest a modification in the conditional compilation statements for importing C libraries. Here is the current code:

#if canImport(Darwin)
import Darwin
#elseif canImport(Glibc)
import Glibc
#elseif canImport(WinSDK)
import WinSDK
#endif

And here is the proposed change:

#if canImport(Darwin)
import Darwin
#elseif canImport(Glibc)
import Glibc
#elseif canImport(Musl)
import Musl
#elseif canImport(WinSDK)
import WinSDK
#endif

This change will allow the framework to dynamically check for the availability of Glibc or Musl, thus supporting environments built with Musl. This would greatly enhance the flexibility and usability of SwifterSwift in various Linux environments.

For further reference on static linking in Linux, you can visit the official Swift documentation: [Static Linux Getting Started](https://www.swift.org/documentation/articles/static-linux-getting-started.html).

Thank you for considering this enhancement.

**Comment by krzyzanowskim** [comment, marked as answer]
thank you. [PR](https://github.com/krzyzanowskim/CryptoSwift/pull/1058) with change merged
